File: array_hashing/217.contains-duplicate.java
package array_hashing;

import java.util.HashSet;

/*
 * @lc app=leetcode id=217 lang=java
 *
 * [217] Contains Duplicate
 */

// @lc code=start
class Solution {
    public boolean containsDuplicate(int[] nums) {
        // Solution By ILham SUrya 15/06/22
        // HashSet Approach Time O(N), Space O(N)
        HashSet<Integer> set = new HashSet<>();
        for (int i : nums) {
            if (!set.add(i))
                return true;
        }
        return false;
    }
}
// @lc code=end

File: array_hashing/golang/217.contains-duplicate.go
/*
 * @lc app=leetcode id=217 lang=golang
 *
 * [217] Contains Duplicate
 */

// @lc code=start
func containsDuplicate(nums []int) bool {
    // HashMap Approach
    // Solution by ilham surya

    // Space: O(n)
	set := make(map[int]bool)

    // Time: O(n)
	for _, num := range nums {
		if _, ok := set[num]; ok {
			return true
		}
		set[num] = true
	}
	return false
}
// @lc code=end

